### PR TITLE
change random strategy slightly, fixes #2565

### LINKED ISSUE #2565: Error in fixed field computation
Seems to be random and observerd here [here](https://github.com/oscar-system/Oscar.jl/actions/runs/5557542180/jobs/10151460991?pr=2510#step:6:1044).

Here is the error. CC: @fieker 
```
-> Testing ./NumberTheory/nmbthy.jl took: 2.527 seconds, 140.23 MB
galois_group: Error During Test at /home/runner/work/Oscar.jl/Oscar.jl/test/NumberTheory/galthy.jl:1
  Got exception outside of a @test
  dnw
  Stacktrace:
    [1] error(s::String)
      @ Base ./error.jl:33
    [2] relative_invariant(G::PermGroup, U::PermGroup; Chain::Nothing)
      @ Oscar.GaloisGrp ~/work/Oscar.jl/Oscar.jl/src/NumberTheory/GaloisGrp/GaloisGrp.jl:2262
    [3] relative_invariant
      @ ~/work/Oscar.jl/Oscar.jl/src/NumberTheory/GaloisGrp/GaloisGrp.jl:2203 [inlined]
    [4] fixed_field(GC::Oscar.GaloisGrp.GaloisCtx{Hecke.qAdicRootCtx}, U::PermGroup, extra::Int64)
      @ Oscar.GaloisGrp ~/work/Oscar.jl/Oscar.jl/src/NumberTheory/GaloisGrp/GaloisGrp.jl:2288
    [5] fixed_field(GC::Oscar.GaloisGrp.GaloisCtx{Hecke.qAdicRootCtx}, U::PermGroup)
      @ Oscar.GaloisGrp ~/work/Oscar.jl/Oscar.jl/src/NumberTheory/GaloisGrp/GaloisGrp.jl:2283
    [6] macro expansion
      @ ~/work/Oscar.jl/Oscar.jl/test/NumberTheory/galthy.jl:9 [inlined]
    [7] macro expansion
      @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
    [8] top-level scope
      @ ~/work/Oscar.jl/Oscar.jl/test/NumberTheory/galthy.jl:3
    [9] include
      @ ./Base.jl:385 [inlined]
   [10] macro expansion
      @ ./timing.jl:368 [inlined]
   [11] include(str::String)
      @ Main ~/work/Oscar.jl/Oscar.jl/test/runtests.jl:76
   [12] (::Base.var"#834#839"{typeof(include)})(r::Base.RefValue{Any}, args::Tuple{String})
      @ Base ./asyncmap.jl:100
   [13] macro expansion
      @ ./asyncmap.jl:234 [inlined]
   [14] (::Base.var"#850#851"{Base.var"#834#839"{typeof(include)}, Channel{Any}, Nothing})()
      @ Base ./task.jl:417
Test Summary: | Pass  Error  Total
galois_group  |    1      1      2
ERROR: LoadError: LoadError: Some tests did not pass: 1 passed, 0 failed, 1 errored, 0 broken.
in expression starting at /home/runner/work/Oscar.jl/Oscar.jl/test/NumberTheory/galthy.jl:1
in expression starting at /home/runner/work/Oscar.jl/Oscar.jl/test/runtests.jl:152
ERROR: LoadError: Package Oscar errored during testing
```

### PR DESCRIPTION
**Pull request by fieker**

### CODE REVIEW

File: src/NumberTheory/GaloisGrp/GaloisGrp.jl [hunk 1]
```diff
@@ -2257,6 +2257,7 @@ function relative_invariant(G, U; Chain::Union{Nothing, <:Vector{<:Tuple{PermGro
       if length(d) == length(T)
         break
       end
+      r = [rand(k) for i=1:degree(G)]
```
Comment (fingolfin) [inline src/NumberTheory/GaloisGrp/GaloisGrp.jl:2260]: `r` is not used, is it? So all this does is to perturb the current state of the default Julia RNG, right?

Comment (thofma) [reply src/NumberTheory/GaloisGrp/GaloisGrp.jl:2260]: 5 lines above

Comment (fieker) [reply src/NumberTheory/GaloisGrp/GaloisGrp.jl:2260]: I need to compare sl-polys. In general I cannot, writing them as mpolys is impossible, so I do the next best thing: In the end I need to prove that two polys are different. So I find random evaluation points and compare the evaluation.

If then eval different, then the polys are different. If they eval is equal I did not lear anything.

In here, the eval poitns are chosen once and for all at the top. However, after a comparison returned equal, I am changing them - as my previous batch might have been dud.

Comment (fingolfin) [reply src/NumberTheory/GaloisGrp/GaloisGrp.jl:2260]: My bad, I missed the use of `r` a few lines above in line 2256

**Review by fingolfin — APPROVED** [review]

**Pull request merged**